libraries/ms-common/inc/spi.h

```diff
@@ -28,5 +28,20 @@ typedef struct {
 // exactly as requested. Please verify that the actual baudrate is within bounds.
 StatusCode spi_init(SpiPort spi, const SpiSettings *settings);
 
+// This method will send |tx_len| bytes from |tx_data| to the spi port |spi|. It will not
+// change the CS line state. The response bytes will be discarded.
+StatusCode spi_transmit(SpiPort spi, uint8_t *tx_data, size_t tx_len);
+
+// This method will receive |rx_len| bytes and place it into |rx_data| from the spi port
+// |spi|. It will not change the CS line state. In order to receive data this method will
+// send the byte specified by the |placeholder| parameter.
+StatusCode spi_receive(SpiPort spi, uint8_t *rx_data, size_t rx_len, uint8_t placeholder);
+
+// This method will set the state of the CS line for a given spi port
+StatusCode spi_set_cs_state(SpiPort spi, GpioState state);
```

Comment:
Would we want a `spi_get_cs_state` as well? Unless there is some other way of determining this I'm unaware of

Reply:
Good catch, I didn't think of that